service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
         return userServiceMapper.convertUserToDto(user);
     }
 
+    @Transactional
     @Override
-    public Long count(){
+    public UserDto signup(UserDto userDto) {
+        User user = userServiceMapper.convertUserFromDto(userDto);
+
+        List<ValidationError> validationErrors = userValidator.validate(user);
+
+        if (!validationErrors.isEmpty()) {
+            throw new InvalidEntityException(User.class);
+        }
+
+        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
+            throw new EntityAlreadyExistsException();
+        }
+
+        String encodedPassword = encoder.encode(user.getPassword());
+        user.setPassword(encodedPassword);
+
+        User savedUser = userRepository.save(user);
+        Optional<User> userById = userRepository.findById(savedUser.getId());
+        if (userById.isEmpty()) {
+            throw new EntityNotFoundException(String.valueOf(savedUser.getId()));
+        }
+        return userServiceMapper.convertUserToDto(userById.get());
+    }
+
+    @Override
+    public UserDto login(UserDto userDto) {
+        String username = userDto.getUsername();
+        String password = userDto.getPassword();
+
```

Comment:
I would not recommend getting used to leaving blank lines. This clearly violates the Java Code Convention. But in some cases it improves the readability of the code. In your case, there is definitely no such need